src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -139,6 +147,15 @@ private void NotifyOnForegroundWorker()
                         {
                             // eat up cancellation
                         }
+                        catch (Exception ex) when (FatalError.ReportWithoutCrash(ex))
+                        {
+                            // The PendingWork callbacks should never throw.  In the case they do we
+                            // must ensure the IAsyncToken implementation is completed.  If it is not
```

Comment:
If this should never throw, then adding the crash shouldn't hurt the product. Since clearly this statement is vacuously true (:smile:) let's go with an easier change I think. If we continue to choose to not crash, then I'd expect you to add tests that this system properly recovers from exceptions being thrown.